src/sdk/object_sdk.js

```diff
@@ -158,6 +165,12 @@ class ObjectSDK {
         }
     }
 
+    // validates requests for non nsfs buckets from nsfs accounts which are allowed_non_fs_buckets 
+    validate_allowed_nsfs_account(account) {
+        if (!account) return true;
```

Comment:
when is account undefined? and why is that allowing access?

Reply:
I added it since we are accessing properties of account, and I changes the logic, now when no account it throws an UNAUTHORIZED error.